src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -526,10 +534,14 @@ public void setDatasetBatchSize(Integer datasetBatchSize) {
     this.datasetBatchSize = datasetBatchSize;
   }
 
-  public Dataset updateDatasetIndex(Integer datasetId, Integer userId, Instant indexDate)
-      throws SQLException {
-    datasetServiceDAO.updateDatasetIndex(datasetId, userId, indexDate);
-    return datasetDAO.findDatasetById(datasetId);
+  protected void updateIfIndexed(Dataset dataset, Integer userId, Instant indexDate) {
+    if (dataset.getIndexedDate() != null) {
+      try {
+        datasetServiceDAO.updateDatasetIndex(dataset.getDatasetId(), userId, indexDate);
```

Comment:
Is this fix incomplete? I would expect to see a call to elasticsearch to actually update the index (which would update the DB), rather than this updating the database record directly.

Reply:
This PR is still in draft - I'm still verifying all of the changes functionally, so this could very well be incomplete.